power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -345,14 +349,20 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     }
 
     // restore the initial values of all components
-    void restore_components(SequenceIdx const& sequence_idx) {
+    void restore_components(SequenceIdxView const& sequence_idx) {
         (restore_component<ComponentType>(sequence_idx), ...);
 
         update_state(cached_state_changes_);
         cached_state_changes_ = {};
     }
-
-    void restore_components(ConstDataset const& update_data) { restore_components(get_sequence_idx_map(update_data)); }
+    void restore_components(std::array<std::reference_wrapper<std::vector<Idx2D> const>, n_types> const& sequence_idx) {
```

Comment:
it's the other way around. we need the reference wrapper so that we can have an array with references to where we get the sequence from, i.e., the local cache for this scenario (if it is not independent for this component type) or the global cache for all scenarios (if it is independent for this component type)

probably is something for the knowledge sharing session

Reply:
This one I didn't really get why we need the reference wrapper, as we can just get reference to the `std::vector` itself. But then please enlighten me in knowledge session. Thanks!

Reply:
short answer: raw reference `&` cannot be default-constructed (cause `*nullptr` is undefined behavior). therefore, `std::array<std::vector<T>&>` is is not allowed unless both this and its contents are `constexpr` created and initialized